### ISSUE TITLE
mrz.traineddata and How to implement OCR-B font?

### ISSUE THREAD

**Issue by nikolacrnicrni** [mercuriete/android-mrz-reader#6]
Can't find in the code how does it implement mrz.traineddata and what is it main role?
Also Is there a way to implement OCR-B font that is used for passports, because I cant find anything about that.

Best Regards

**Comment by mercuriete** [comment]
Hi @nikolacrnicrni 
I had a lot of problems about traineddata.

You can check the documentation in this url:

https://github.com/tesseract-ocr/tesseract/wiki/Data-Files#format-of-traineddata-files

This project is a spare time project so I will update it on "best effort".

Anyway, in my investigations... the best performance is using generic trained data from english.

PS: for training tesseract i used this app -> https://www.youtube.com/watch?v=HgeksbqH0ng

**Comment by faiqmustaqeem** [comment]
@nikolacrnicrni , did you find any solution ?

**Comment by nikolacrnicrni** [comment]
No, at the end I used .so file made in C++ and implemented in android project to make scaning.

**Comment by mercuriete** [comment]
@nikolacrnicrni sorry if I couldn't help you.

I didn't have any time to check the code.
If you want to share your findings about trained data here you are welcome.

Thanks for checking this repository :)

**Comment by JakeFinch** [comment]
@nikolacrnicrni , can you please share details or sample code how you implemented scanning using .so file?

**Comment by faiqmustaqeem** [comment]
@nikolacrnicrni , can you please share details which .so file you used and was it working perfect?

**Comment by nikolacrnicrni** [comment]
@faiqmustaqeem I'm sorry but no. I am now selling that .so file because I lost months of developing. It is created with OpenCV.

**Comment by mercuriete** [comment]
@nikolacrnicrni 
Remember the license of the .so you are selling.
Remember if you sell a opensource project you have to comply with the license and provide de source code.

**Comment by nikolacrnicrni** [comment]
@mercuriete Yeah I know. But I started the project from begining with my code with OpenCV library.

> OpenCV was built to provide a common infrastructure for computer vision applications and to accelerate the use of machine perception in the commercial products. Being a BSD-licensed product, OpenCV makes it easy for businesses to utilize and modify the code.

https://github.com/opencv/opencv/blob/master/LICENSE

**Comment by mercuriete** [comment]
@nikolacrnicrni Nice!
This is the awesome of good licenses!
I hope you are doing well in your bussiness 👍 

Do you had problems with zero "0" and uppercase "o"?

I remembered some time ago trying to read European Visas, there was a field outside the checksum that was difficult to read with any computer vision framework.
So in the end we fixed by hand because we knew it should be a number.

**Comment by faiqmustaqeem** [comment]
@nikolacrnicrni , i want to buy, can you please share demo app apk ?

**Comment by nikolacrnicrni** [comment]
@faiqmustaqeem I have sent it to your email.
